### PR TITLE
UF-6687 & UF-6689

### PR DESCRIPTION
**Pull request by AndreSoderlund**
API changes to match version 1.1 of emailsender. Some mapping that also needed to be adjusted along with some tests.

### CODE REVIEW

**Review by MickeWallner — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
         <groupId>se.sundsvall.dept44</groupId>
         <artifactId>dept44-service-parent</artifactId>
         <version>4.1</version>
-        <relativePath />
+        <relativePath/>
     </parent>
     <groupId>se.sundsvall.messaging</groupId>
     <artifactId>api-service-messaging</artifactId>
-    <version>4.0</version>
+    <version>4.1</version>
```
Comment (jklingst) [inline pom.xml:14]: 4.2.0?

File: pom.xml [hunk 2]
```diff
@@ -21,7 +22,8 @@
         <hypersistence-utils.version>3.7.0</hypersistence-utils.version>
         <vavr.version>0.10.4</vavr.version>
         <archunit.version>1.2.1</archunit.version>
-        <generated-sources-path>${project.build.directory}/generated-sources</generated-sources-path>
+        <generated-sources-path>${project.build.directory}/generated-sources
+        </generated-sources-path>
```
Comment (jklingst) [inline pom.xml:26]: Why an additional newline?

File: pom.xml (hunk 1 above)
Comment (AndreSoderlund) [reply pom.xml:14]: I get using (X.Y.Z), but why skip 4.1.0?

File: src/main/java/se/sundsvall/messaging/integration/emailsender/EmailSenderIntegrationMapper.java [hunk 3]
```diff
@@ -2,38 +2,38 @@
...
 class EmailSenderIntegrationMapper {
 
-    SendEmailRequest toSendEmailRequest(final EmailDto dto) {
-        if (dto == null) {
-            return null;
-        }
+	private EmailSenderIntegrationMapper() {
+	}
+
+	public static SendEmailRequest toSendEmailRequest(final EmailDto dto) {
```
Comment (jklingst) [inline src/main/java/se/sundsvall/messaging/integration/emailsender/EmailSenderIntegrationMapper.java:14]: Package-protected

File: src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java [hunk 4]
```diff
@@ -46,7 +47,10 @@ public record EmailRequest(
         Sender sender,
 
         @ArraySchema(schema = @Schema(implementation = Attachment.class))
-        List<@Valid Attachment> attachments) {
+        List<@Valid Attachment> attachments,
+
+        @Schema(description = "Headers")
+        Map<Header, List<String>> headers) {
```
Comment (jklingst) [inline src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java:53]: Should there be some kind of constraint(s) on these?

File: pom.xml (hunk 1 above)
Comment (jklingst) [reply pom.xml:14]: Since 4.1.0 doesn't exist at all. SemVer was (re)introduced with 4.2.0 (https://github.com/Sundsvallskommun/dept44/pull/244)

Comment (jklingst) [reply pom.xml:14]: Yes

File: src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java [hunk 5]
```diff
@@ -21,77 +23,89 @@
...
+
+	@Valid
+	@Schema(description = "Sender")
+	Sender sender,
+
+	@ArraySchema(schema = @Schema(implementation = Attachment.class))
+	List<@Valid Attachment> attachments,
+
+	@Schema(description = "Headers")
+	Map<Header, List<@Pattern(regexp = "^<.{1,1000}@.{1,1000}>$", message = "Message-ID need to begin with '<' contain '@' and end with '>'") String>> headers) {
```
Comment (josc04) [inline src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java:54]: Just curious, won't this validation trigger and reject payload if client chooses to send other headers than message-id? I.e. if client chooses to send reply_to, wont this validation reject that header?

If the pattern is valid for all header values, maybe change the error message to a more "general" one, for example `header values must begin with '<', contain '@' and end with '>'`?

Comment (AndreSoderlund) [reply src/main/java/se/sundsvall/messaging/api/model/request/EmailRequest.java:54]: The three headers, In-Reply-To, References and Message-ID should all contain either one or more 'Message-ID's.

**Review by josc04 — APPROVED** [review]

**Pull request merged**